dv/uvm/icache/doc/ibex_icache_dv_plan.md

```diff
@@ -1,99 +1,142 @@
-<!-- TODO -->
-* IBEX_ICACHE IOs
+* Clock and reset interface ([`vendor/lowrisc_ip/common_ifs`](https://github.com/lowRISC/ibex/tree/master/vendor/lowrisc_ip/common_ifs))
+* Core interface ([`dv/uvm/icache/dv/ibex_icache_core_agent/ibex_icache_core_if.sv`](https://github.com/lowRISC/ibex/blob/master/dv/uvm/icache/dv/ibex_icache_core_agent/ibex_icache_core_if.sv))
+* Memory interface ([`dv/uvm/icache/dv/ibex_icache_mem_agent/ibex_icache_mem_if.sv`](https://github.com/lowRISC/ibex/blob/master/dv/uvm/icache/dv/ibex_icache_mem_agent/ibex_icache_mem_if.sv))
 
 ### Common DV utility components
-The following utilities provide generic helper tasks and functions to perform activities that are common across the project:
-* [dv_utils_pkg]({{< relref "vendor/lowrisc_ip/dv_utils/README.md" >}})
+
+TBD
 
 ### Compile-time configurations
-[list compile time configurations, if any and what are they used for]
+
+TBD
 
 ### Global types & methods
 All common types and methods defined at the package level can be found in
 `ibex_icache_env_pkg`. Some of them in use are:
-```systemverilog
-[list a few parameters, types & methods; no need to mention all]
-```
-
-###  IBEX_ICACHE Agent
-[Describe here or add link to its README]
-###  IBEX_MEM_INTF_SLAVE Agent
-[Describe here or add link to its README]
-
-### UVC/agent 1
-[Describe here or add link to its README]
-
-### UVC/agent 2
-[Describe here or add link to its README]
 
-
-### Reference models
-[Describe reference models in use if applicable, example: SHA256/HMAC]
+TBD
 
 ### Stimulus strategy
 #### Test sequences
-All test sequences reside in `dv/uvm/icache/dv/env/seq_lib`.
-The `ibex_icache_base_vseq` virtual sequence is extended from `cip_base_vseq` and serves as a starting point.
-All test sequences are extended from `ibex_icache_base_vseq`.
-It provides commonly used handles, variables, functions and tasks that the test sequences can simple use / call.
-Some of the most commonly used tasks / functions are as follows:
-* task 1:
-* task 2:
+
+All test sequences reside in [`dv/uvm/icache/dv/env/seq_lib`](https://github.com/lowRISC/ibex/tree/master/dv/uvm/icache/dv/env/seq_lib).
+
+TBD
 
 #### Functional coverage
-To ensure high quality constrained random stimulus, it is necessary to develop a functional coverage model.
-The following covergroups have been developed to prove that the test intent has been adequately met:
-* cg1:
-* cg2:
+
+TBD
 
 ### Self-checking strategy
 #### Scoreboard
-The `ibex_icache_scoreboard` is primarily used for end to end checking.
-It creates the following analysis ports to retrieve the data monitored by corresponding interface agents:
-* analysis port1:
-* analysis port2:
-<!-- explain inputs monitored, flow of data and outputs checked -->
+The `ibex_icache_scoreboard` is used to check that instructions fetched from the cache lie at the correct addresses and contain the correct data.
+It has two analysis ports: one for the memory agent and another for the core agent.
```

Comment:
If the `busy_o` agent is not going to communicate with the scoreboard, is there a need for a full agent to be developed around this one signal?

Reply:
I wrote that text before adding the `busy_o` agent above but then forgot to change it: good catch! However, I agree that we don't need the extra agent, so maybe I should just claim I was being prescient :-D.

Reply:
Yes, understood. Since I'm folding this single pin into another agent, I'll just add it to the core <-> icache interface. I think that's even a principled decision (as well as being convenient!) because it is indeed part of the interface between the cache and the core.